ocean_provider/routes/decrypt.py

```diff
@@ -72,23 +72,27 @@ def _decrypt(
     # Check if given chain_id matches Provider's chain_id
     web3 = get_web3()
     if web3.chain_id != chain_id:
-        return error_response(f"Unsupported chain ID", 400, logger)
+        return error_response("Unsupported chain ID", 400, logger)
 
     # Check if decrypter is authorized
     authorized_decrypters = get_config().authorized_decrypters
     logger.info(f"authorized_decrypters = {authorized_decrypters}")
     if authorized_decrypters and decrypter_address not in authorized_decrypters:
-        return error_response(f"Decrypter not authorized", 403, logger)
+        return error_response("Decrypter not authorized", 403, logger)
+
+    factory_contract = get_data_nft_factory_contract(web3)
+    if factory_contract.caller.erc721List(data_nft_address) != data_nft_address:
+        return error_response(
+            "Asset not deployed by the data NFT factory.", 400, logger
+        )
```

Comment:
Let's add this as a function in `data_nft_factory.py`.

```python
def is_nft_deployed_from_factory(nft_address: str) -> bool:
    """Check if NFT is deployed from the factory."""
    data_nft_factory = get_data_nft_factory_contract(web3)
    return data_nft_factory.caller.erc721List(nft_address) == nft_address
```

Then the usage here would become

```python
if not is_nft_deployed_from_factory(data_nft_address):
        return error_response(
            "Asset not deployed by the data NFT factory.", 400, logger
        )
```

Also, as a side note, this comment showed me we're missing this method (and the corresponding one for datatokens) in ocean.py